laws/shared/src/main/scala/algebra/laws/RunSerializationLaws.scala

```diff
@@ -0,0 +1,9 @@
+package algebra.laws
+
+sealed trait SerializationCheck
```

Comment:
yeah, that makes sense for java serialization. It's odd to have this annotation as well as the java.io.Serializable extension, since the only override you're going to do is mark a class NOT serializable, with `IsNotSerializable[M]`, while the class is still going to be serializable as far as java's concerned.

Letting the class itself opt out seems like the right way to go. Tests and users that want to store instances can have a better check then the current reflection option.